pkg/credentials/common/constants.go

```diff
@@ -0,0 +1,21 @@
+package common
+
+// Secret key constants used in credential files,
+// so as to avoid reliance on corev1.Secret.
```

Comment:
Since we still need to have `corev1` in our `go.mod`, could you explain what is the issue with that?
Would only importing `corev1` to get these constants help?

Since these are from a v1 API, they won't change, so it should be fine, but I would like to understand what's the benefit of doing this.

Reply:
`corev1` in go.mod doesn't mean it is "imported" and thus present in the binary (here `entrypoint`). The "problem" with `corev1` is that it brings a bunch of dependencies that bring crypto/tls imports and thus making FIPS are to get for `entrypoint`. The goal for `entrypoint` is to make sure it doesn't communicate and doesn't import any crypto/tls libraries (that we know it doesn't need for 99% of cases).

Reply:
+1 @vdemeester 
we are ensuring that corev1 and deps (especially the crypto/tls) don't end up in the entrypoint. The functionality required to get the job done for Matching annotations for Secrets requires these constants. Instead of importing these files from corev1, we are putting whatever subset of code required for parsing Secrets we need within the codebase.